app/src/main/java/de/danoeh/antennapod/fragment/preferences/AutoDownloadPreferencesFragment.java

```diff
@@ -175,10 +188,77 @@ private void buildEpisodeCleanupPreference() {
     }
 
     private void setSelectedNetworksEnabled(boolean b) {
+        if (showPermissionRequestPromptOnAndroid10IfNeeded(b)) {
+            return;
+        }
+
         if (selectedNetworks != null) {
             for (Preference p : selectedNetworks) {
                 p.setEnabled(b);
             }
         }
     }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode != permissionRequestCode) {
+            return;
+        }
+        if (permissions.length > 0 && permissions[0].equals(requestedPermission) &&
+                grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            buildAutodownloadSelectedNetworksPreference();
+        }
+    }
+
+    private boolean showPermissionRequestPromptOnAndroid10IfNeeded(boolean wifiFilterEnabled) {
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.P) {
+            return false;
+        }
+
+        // Cases Android 10(Q) or later
+
+        final PreferenceScreen prefScreen = getPreferenceScreen();
```

Comment:
This is only used one single time and only in the if below. You can just use the method there.